src/test/java/serenitylabs/tutorials/vetclinic/collections/katas/WhenBookingPetsIntoAPetHotel.java

```diff
@@ -1,54 +1,186 @@
+        Pet tommy = Pet.dog().named("Tommy");
+
+        //WHEN
+        when(vikrant).attemptsTo(CheckIn.aPet(tommy).in(petHotel));
+
+        //THEN
+        then(vikrant).should(seeThat(TheWaitingGuest.isOnWaitingListIn(petHotel), hasItem(tommy)));
     }
 
     @Test
     public void pets_on_the_waiting_list_should_be_added_to_the_hotel_when_a_place_is_freed() throws Exception {
+        //GIVEN
+        PetHotel petHotel = APetHotel.with(19).checkedIn();  //adding random 20 pets as of now
+        Pet tommy = Pet.dog().named("Tommy");
+        Pet goldy = Pet.fish().named("Goldy the Fish");
+
+        //WHEN
+        when(vikrant).attemptsTo(CheckIn.aPet(tommy).in(petHotel)); //VIKRANT trying to checkin
+
+        when(john).attemptsTo(CheckIn.aPet(goldy).in(petHotel)); //JOHN trying to checkin
+
+        when(vikrant).attemptsTo(CheckOut.aPet(tommy).in(petHotel)); //VIKRANT trying to checkout
+
+        //THEN
+        then(vikrant).should(seeThat(TheRegisteredGuest.registeredIn(petHotel), hasItem(goldy)));
     }
 
 
     @Test
     public void pets_on_the_waiting_list_should_be_admitted_on_a_first_come_first_served_basis() throws Exception {
+        //GIVEN
+        PetHotel petHotel = APetHotel.with(19).checkedIn();  //adding random 20 pets as of now
+        Pet tommy = Pet.dog().named("Tommy");
+        Pet goldy = Pet.fish().named("Goldy the Fish");
+        Pet anabelle = Pet.cat().named("Anabelle the Cat");
+
+        //WHEN
+        when(vikrant).attemptsTo(CheckIn.aPet(tommy).in(petHotel)); //VIKRANT trying to checkin
+
+        when(john).attemptsTo(CheckIn.aPet(goldy).in(petHotel)); //JOHN trying to checkin
+
+        when(vikrant).attemptsTo(CheckIn.aPet(anabelle).in(petHotel)); //VIKRANT trying to checkin
+
+        when(vikrant).attemptsTo(CheckOut.aPet(tommy).in(petHotel)); //VIKRANT trying to checkout
+
+        //THEN
+        then(vikrant).should(seeThat(TheRegisteredGuest.registeredIn(petHotel), hasItem(goldy)),
+                seeThat(TheWaitingGuest.isOnWaitingListIn(petHotel), hasItem(anabelle)));
+    }
+
+    @Test
+    public void should_be_able_to_obtain_a_booking_confirmation_when_we_check_in_a_pet() throws Exception {
+        //GIVEN
+        Pet tommy = Pet.dog().named("Tommy");
+
+        //WHEN
+        when(vikrant).attemptsTo(CheckIn.aPet(tommy).in(petHotel));
+
+        //THEN
+        then(vikrant).should(seeThat(TheBookingStatus.of(tommy).isConfirmed(), is(true)));
     }
 
+    @Test
+    public void should_notify_owner_that_the_hotel_is_full() throws Exception {
+        //GIVEN
+        PetHotel petHotel = APetHotel.with(19).checkedIn();
+        Pet tommy = Pet.dog().named("Tommy");
+        Pet goldy = Pet.fish().named("Goldy");
+
+        //WHEN
+        when(vikrant).attemptsTo(CheckIn.aPet(tommy).in(petHotel), CheckIn.aPet(goldy).in(petHotel));
+
+        //THEN
+        then(vikrant).should(seeThat(TheWaitingStatus.of(goldy).isConfirmed(), is(false)));
+    }
```

Comment:
This test is missing an edge case. If the hotel is full, the booking status should fail, e.g.
```
    @Test
    public void should_notify_owner_that_the_hotel_is_full() throws Exception {
        //GIVEN
        PetHotel petHotel = APetHotel.with(19).checkedIn();
        Pet tommy = Pet.dog().named("Tommy");
        Pet goldy = Pet.fish().named("Goldy");

        //WHEN
        when(vikrant).attemptsTo(CheckIn.aPet(tommy).in(petHotel), CheckIn.aPet(goldy).in(petHotel));

        //THEN
        then(vikrant).should(seeThat(TheBookingStatus.of(goldy).isConfirmed(), is(false)));
        and(vikrant).should(seeThat(TheWaitingStatus.of(goldy).isConfirmed(), is(false)));
    }
```